src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1575,6 +1576,8 @@ class Simplex_tree {
    * bound. If you care, you can call `dimension()` to recompute the exact dimension.
    */
   bool prune_above_filtration(Filtration_value filtration) {
+    if (!std::isfinite(filtration))
```

Comment:
`prune_above_filtration(-INFINITY)` should remove (almost?) everything, I don't think `isfinite` is right.